### ISSUE TITLE
Files duplicated when syncing across devices

### ISSUE THREAD

**Issue by Nenuvar** [stravo1/obsidian-gdrive-sync#46]
I recently installed this plugin so that I could sync across several devices: Chromebook, Android, Windows PC and iPad. The initial syncing went great! At this point I only had the "Welcome" note in the vault and did some minor changes to test the sync. 

After a couple of days I had expanded on my vault with some notes, mostly written on my Windows PC. I noticed that several of the files and folders had been duplicated on both my Android phone and Chromebook. I ended up having to delete the vaults from there and reinstall them from my GitHub backup. The duplication has presisted, and I regularly encounter an error timeout of "too many errors in a minute". I get this error mostly when I am doing many edits on one device (like my Windows PC) and then opening another device later on.  

I use the plugin Folder Notes by LostPaul and thought maybe the folder note setup causes the syncing problems, but I'm not sure. 

Also, it seems to be downloading my entire vault each time I open the app. I have a pretty small vault right now (around 60 files), but I plan to expand on it pretty extensively later on. Downloading hundreds of files each time I open the app does not seem efficient, and I can't see a way it will run smoothly. Is there a way to only download changes I made, not the entire vault?

Currently I am not using the plugin, as it keeps duplicating my changes. I would really like it to work, as I love the idea and I would really like a free sync alternative for iOS.

**Comment by stravo1** [comment]
hello! thanks for reporting this issue!

can you explain what you mean by duplication? are there multiple copies of the same file? if so are they renamed? also can you check whether the duplicated files both have the `lastSync` tag?

the plugin you mentioned might also be the problem, if possible can you test this plugin on a vault which does not have that plugin installed?

also the plugin does not download the entire vault, it should only download the missing files, i.e, files which are there in the Drive but not on the local device.

**Comment by Nenuvar** [comment]
There are multiple copies of files with the same name in my vault. They all
have different "last sync" tags, but the same title. I must say, I did a
lot of changes in a short amount of time, so that has maybe interrupted the
sync.

I will try setting up the vault from scratch to see if the problem
persists!

I set up an entirely new vault with only the Google Drive Sync Plugin
installed (and the welcome.md file) on my Windows PC. Then I downloaded the
Google Drive vault file to my Android phone and set up the vault there. I
tried creating a bunch of notes on both devices and they synced quickly and
beautifully.

Then I tried creating some folders. And this is where the issue is. I first
created a folder called "Windows folder" on my PC that contained a note
with the title "Number 1". This synced correctly to my Android. Then I
renamed the folder "Not Windows anymore". On my Android I got a new folder
called "Not Windows anymore" with the "Number 1" note in it. The issue is
that the "Windows folder" is empty, but still present on my phone (even
though it is not present on my PC or Google Drive). I tried the "download
all" option on my phone, but it didn't fix things. I also tried "upload
all" from my Windows PC, then "download all" on Android, but that didn't
work either.

So it seems the note in the renamed folder got moved to a new folder with
the new name, but the old folder remained. Is there a fix for this?

Thank you so much for responding so quickly, by the way!

Best wishes,

Oliver

**Comment by stravo1** [comment]
unfortunately no, folders are tracked only if they have notes in them, so empty folders are not tracked by the plugin. this should be mentioned on the README. thank you for reporting this.

**Comment by Nenuvar** [comment]
I actually saw that in the readme, but didn't think it would treat a
renamed folder as a new one. Good to know! I'll be careful with naming my
folders from now on ;)

Oh happy days, I can use the plugin again!

**Comment by stravo1** [comment]
if the problem is fixed, we can close this issue. but what about the duplicate files?

**Comment by Nenuvar** [comment]
Yes it is fixed, thank you!

Well, sometimes the file was both in the original folder and the renamed
one, making a duplicate. So I would have a "Number 1" note in both the
"Windows folder" and the "Not Windows anymore" folder. When I deleted the
folder from my PC, only the note was deleted on my Android. So I had to
manually delete the empty folder from Android, even though I deleted it
from my PC. I guess it's because it cannot sync folders without notes.

What I found works pretty well is using the Folder Notes plugin and set it
so that it creates a folder note whenever I create a folder. This
automatically syncs since there is a note associated with the folder.

Thanks again!
